include/deal.II/lac/sparse_matrix.h

```diff
@@ -2377,6 +2384,35 @@ SparseMatrix<number>::end (const size_type r)
 }
 
 
+template <typename number>
+template <class StreamType>
+void SparseMatrix<number>::print_matrix_market(StreamType &out,
+					       const number threshold) const
+{
+  Assert(cols != 0, ExcNotInitialized());
+  Assert(val != 0, ExcNotInitialized());
+  Assert( threshold > 0, dealii::ExcMessage("Negative threshold!") ); 
+
+  //Print the header
+  out << "%%MatrixMarket matrix coordinate real general\n";
+  const auto nnz = n_actually_nonzero_elements(threshold);
+  out << m() << ' ' << n() << ' ' << nnz << '\n';
+
+  //Print the body
+  for(unsigned int i = 0; i < m(); ++i)
+    for(auto it = begin(i); it != end(i); ++it)
```

Comment:
currently we do not require c++11 for deal.II (it's planned to switch in future release). Would you mind avoid using `auto` so that the definition of the function compiles without C++11?